core/sis-referencing/src/test/java/org/apache/sis/geometry/CoordinateFormatTest.java

```diff
@@ -188,4 +193,76 @@ public void testParseGeographic4D() throws ParseException {
                          e.getLocalizedMessage());  // In the language specified at CoordinateFormat construction time.
         }
     }
+
+    @Test
+    public void testCreatesCoordinateFormatTaking2ArgumentsAndCallsClone() {
+        Locale locale = Locale.CANADA;
+        CoordinateFormat coordinateFormat = new CoordinateFormat(locale, null);
+        CoordinateFormat coordinateFormatTwo = coordinateFormat.clone();
+
+        assertNotSame(coordinateFormatTwo, coordinateFormat);
+    }
+
+    @Test
+    public void testParseAndFormatTaking2Arguments() throws IOException, ParseException {
+        CoordinateFormat coordinateFormat = new CoordinateFormat();
+        DirectPosition2D directPosition2D = new DirectPosition2D();
+        StringBuffer stringBuffer = new StringBuffer();
+        coordinateFormat.format(directPosition2D, stringBuffer);
+        ParsePosition parsePosition = new ParsePosition(0);
+        DirectPosition directPosition = coordinateFormat.parse(stringBuffer, parsePosition);
+
+        assertEquals(2, directPosition.getDimension());
+        assertEquals(2, directPosition.getCoordinate().length);
+        assertEquals(0.0, directPosition.getCoordinate()[0], 0.01);
+        assertEquals(0.0, directPosition.getCoordinate()[1], 0.01);
+    }
+
+    @Test
+    public void testParseAndSetDefaultCRSAndParse() throws ParseException {
```

Comment:
What this method is testing compared to other methods is the capability to test parsing from a position different than 0.  The use of `setDefaultCRS` has been tested by other methods, so it is not new in this method. I propose to rename this method `testParseFromOffset`, which is the new thing compared to other methods.